core/modules/comment/comment.theme.inc

```diff
@@ -104,9 +104,14 @@ function template_preprocess_comment(&$variables) {
 
   $uri = $comment->uri();
   $uri['options'] += array('attributes' => array('class' => array('permalink'), 'rel' => 'bookmark'));
-
-  $variables['title'] = l($comment->subject, $uri['path'], $uri['options']);
+  
   $variables['permalink'] = l(t('Permalink'), $uri['path'], $uri['options']);
+  // Check to see if title should be hidden
+  if ($node_type->settings['comment_subject_options'] == COMMENT_TITLE_HIDDEN) {
+    $uri['options']['attributes']['class'][] = 'hide_comment_title';
```

Comment:
CSS classes usually should use hyphens, not underscores.

Reply:
I should know better. Fixed.